app/src/main/java/com/rkm/tasky/network/interceptor/TaskyAuthorizationInterceptor.kt

```diff
@@ -1,38 +1,39 @@
 package com.rkm.tasky.network.interceptor
 
-import com.rkm.tasky.BuildConfig
+import RequestType.Companion.fromRequest
 import com.rkm.tasky.network.authorization.abstraction.AuthorizationManager
-import com.rkm.tasky.network.datasource.TaskyRemoteDataSource.RequestType
-import com.rkm.tasky.network.datasource.TaskyRemoteDataSource.RequestType.Companion.fromRequest
 import com.rkm.tasky.util.result.Result
+import dagger.Lazy
 import kotlinx.coroutines.runBlocking
 import okhttp3.Interceptor
 import okhttp3.Response
 import javax.inject.Inject
 
-class TaskyInterceptor @Inject constructor(
-    private val manager: AuthorizationManager,
+class TaskyAuthorizationInterceptor @Inject constructor(
+    private val lazyManager: Lazy<AuthorizationManager>,
 ): Interceptor {
 
-    private val apiHeader = "x-api-key"
     private val authHeader = "Authorization"
     private val authKey = "Bearer"
+    private lateinit var manager: AuthorizationManager
 
     override fun intercept(chain: Interceptor.Chain): Response {
 
-        //If null, no prior login.
-        val authInfo = runBlocking {
-            manager.getSessionInfo()
-        }
+        manager = lazyManager.get()
 
         val request = chain.request()
-        val newRequest = request.newBuilder()
-            .addHeader(apiHeader, BuildConfig.API_KEY)
 
-        if(fromRequest(request) == RequestType.AUTHORIZATION && authInfo != null) {
-            newRequest.addHeader(authHeader, "$authKey ${authInfo.accessToken}")
+        if(fromRequest(request) == RequestType.AUTHENTICATION) {
+            return chain.proceed(request)
         }
 
+        val authInfo = runBlocking { manager.getSessionInfo() }
+
+        authInfo ?: return chain.proceed(request)
```

Comment:
Could it make sense to inline that call in the previous line, so you don't need to put that random `authInfo` here in your code that is just for the sake of a null check?